### ISSUE TITLE
grids in /g/data/ik11 don't have the same format

### ISSUE THREAD

**Issue by navidcy** [COSIMA/access-om2#233]
<img width="963" alt="Screen Shot 2021-02-04 at 9 15 51 pm" src="https://user-images.githubusercontent.com/7112768/106878579-39c06a00-672e-11eb-8461-4d0f4400a949.png">
<img width="981" alt="Screen Shot 2021-02-04 at 9 15 59 pm" src="https://user-images.githubusercontent.com/7112768/106878593-3c22c400-672e-11eb-94cd-6172c8fbc5dc.png">

**BUT**

<img width="993" alt="Screen Shot 2021-02-04 at 9 16 06 pm" src="https://user-images.githubusercontent.com/7112768/106878614-40e77800-672e-11eb-8649-315e3672e0cf.png">

**Comment by aekiss** [comment]
I've been able to make do with a workaround for `area_t` (see `loaddata` in `ice_maps.ipynb` [here](https://github.com/aekiss/ice_analysis/tree/7599d67a91957814250446e81d90f4cac342317a), which uses grid paths defined in `exptdata.py` [here](https://github.com/aekiss/ice_analysis/blob/7599d67a91957814250446e81d90f4cac342317a/exptdata.py#L77)), but it would be better to avoid this kludge and fix the 0.1deg grid files.

**Comment by navidcy** [comment]
I can do hocus-pocus to go around it but we should fix it. It makes it cumbersome to compare things across the three resolutions.

(Unfortunately I don't have any idea how to produce these files so when I say "we should fix it" in principle I just hope somebody else will do :()

**Comment by aekiss** [comment]
Agreed, and I also don't know how to do it. I'm guessing @aidanheerdegen does?

FYI I'm abstracting away these cumbersome differences by accessing the data via `exptdata.py` and `loaddata`, so I have a uniform interface to all resolutions and cycles

**Comment by navidcy** [comment]
hm... can you elaborate on this (eg point to a notebook)? Are these methods in cosima-recipes?

**Comment by aekiss** [comment]
see https://github.com/aekiss/ice_analysis/

**Comment by rmholmes** [comment]
Stop me if I'm wrong, but all of our runs include an `ocean_grid.nc` file in the output directories. These have all the variables you should need. Why not just use those?

E.g. for the 1/10th-degree:
`/g/data/ik11/outputs/access-om2-01/01deg_jra55v13_ryf9091/output***/ocean/ocean_grid.nc`

**Comment by navidcy** [comment]
Because those files they have nans for land in geolon, geolat and this creates trouble in plotting or interpolation.

**Comment by navidcy** [comment]
> see `ice_maps.ipynb` in https://github.com/aekiss/ice_analysis/

Seems very complicated... :(

**Comment by navidcy** [comment]
> > see `ice_maps.ipynb` in https://github.com/aekiss/ice_analysis/
> 
> Seems very complicated... :(

Don't bother though. I managed to do what I needed. Just wanted to point out the asymmetry in the grid files.

**Comment by aidanheerdegen** [comment]
Seems @AndyHoggANU made them:
```bash
$ ls -l
total 514020
-rw-r--r-- 1 amh157 ik11  44099122 Apr 12  2020 Antarctic_slope_contour_1000m.npz
-rw-r----- 1 amh157 ik11 466576179 Mar 23  2020 ocean_grid_01.nc
-rw-r----- 1 amh157 ik11  14521988 Mar 20  2020 ocean_grid_025.nc
-rw-r----- 1 amh157 ik11   1146336 Apr  6  2020 ocean_grid_10.nc
```

**Comment by navidcy** [comment]
deep inside we all know @AndyHoggANU's fault?
@AndyHoggANU, can you comment on this asymmetry or recreate the 0.10 deg file to have the same structure as the other ones?

**Comment by AndyHoggANU** [comment]
I just copied them there as a hack. @russfiedler created the 01deg one:

// global attributes:
		:history = "Fri Mar 20 16:34:09 2020: ncrename -v x_T,geolon_t -v x_C,geolon_c -v y_T,geolat_t -v y_C,geolat_c /g/data/v14/pas548/cosima/enkf/grid/old_grid_spec.nc /scratch/v45/raf599/old_grid_spec.nc" ;

**Comment by aidanheerdegen** [comment]
Old school the way to get a grid without missing values was to run the model for a time step unmasked. 

The ice grid has all the lat and lon unmasked, so that is an option too.

**Comment by russfiedler** [comment]
That looks like a file created with a tool that I created to generate info for our assimilation suite which wants info in the form of older style grid_spec.nc. files https://github.com/russfiedler/grid_spec_tools   Should work for any mosaic style set up.

**Comment by navidcy** [comment]
I've been running into more troubles with this asymmetry...

I can fix it if nobody is willing to put their hand up! But I have no idea how? Could you enlighten me? (I don't know Fortran nor I can run Fortran though...)

**Comment by aekiss** [comment]
we are working on fixing this here: https://github.com/COSIMA/cosima-recipes/issues/212
